File: solutions/cpp/Swapping Nodes in a Linked List.cpp
#include <vector>
using namespace std;
struct ListNode
{
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};

class Solution
{
public:
    ListNode *swapNodes(ListNode *head, int k)
    {
        vector<int> v;
        ListNode *temp = head, *current = head;
        int i = 0, count = 0;
        while (temp)
        {
            v.push_back(temp->val);
            count++;
            temp = temp->next;
        }
        swap(k - 1, count - k, v);
        while (current)
        {
            current->val = v[i];
            i++;
            current = current->next;
        }
        return head;
    }

    void swap(int i, int j, vector<int> &v)
    {
        int temp = v[i];
        v[i] = v[j];
        v[j] = temp;
    }
};

File: solutions/python/Swapping Nodes in a Linked List.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
class Solution:
    def swapNodes(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        [temp,current,i,count,arr]=[head,head,0,0,[]]
        while temp:
            arr.append(temp.val)
            temp=temp.next
            count=count+1
        arr[k-1],arr[count-k]=arr[count-k],arr[k-1]
        while current:
            current.val=arr[i]
            i=i+1
            current=current.next;
        return head;